src/components/Dishes.js

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchDishes } from 'redux/dishes/dishesSlice';
+import { fetchRecipe } from 'redux/recipe/recipeSlice';
+import { useNavigate } from 'react-router-dom';
+import { FaRegThumbsUp } from 'react-icons/fa';
+import './styles/Dishes.css';
+
+const Dishes = () => {
+  const dispatch = useDispatch();
+  const { dishes, status, error } = useSelector((state) => state.dishes);
+  const [selectedDish, setSelectedDish] = useState(null);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [likes, setLikes] = useState(() => {
+    const storedLikes = JSON.parse(localStorage.getItem('likes')) || {};
+    return storedLikes;
+  });
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    dispatch(fetchDishes());
+  }, [dispatch]);
+
+  useEffect(() => {
+    localStorage.setItem('likes', JSON.stringify(likes));
+  }, [likes]);
+
+  const handleDetailClick = (id) => {
+    if (id === selectedDish) {
+      return;
+    }
+    dispatch(fetchRecipe(id));
+    setSelectedDish(id);
+    navigate(`/recipe?id=${id}`);
+  };
+
+  const handleSearchInputChange = (event) => {
+    setSearchQuery(event.target.value);
+  };
+
+  const handleLikeClick = (id) => {
+    setLikes((prevLikes) => {
+      const newLikes = { ...prevLikes };
+      newLikes[id] = (newLikes[id] || 0) + 1;
+      return newLikes;
+    });
+  };
+
+  const filteredDishes = dishes.filter((dish) => {
+    const dishName = dish.strMeal.toLowerCase();
+    const dishLetter = dish.strMeal.charAt(0).toLowerCase();
+    const query = searchQuery.toLowerCase();
+    return dishName.includes(query) || dishLetter === query;
+  });
+
+  if (status === 'loading') {
+    return <div>Loading...</div>;
+  }
+
+  if (status === 'failed') {
+    return (
+      <div>
+        Error:
+        {error}
+      </div>
+    );
+  }
+
+  return (
+    <main>
+      <div className="search-bar">
+        <input
+          className="search-term"
```

Comment:
- [ ] [OPTIONAL] Great implementation with the search 👍🏽 . However, I will recommend that the background color/image takes the full width of the screen during a search. This will enhance a good user interface and experience.
### Your Design
<img width="403" alt="Screenshot 2023-04-27 at 18 24 41" src="https://user-images.githubusercontent.com/58692590/234949459-13d5cf08-5cfa-4de3-a4dd-94d6e1b5149c.png">

Reply:
Hi @Whoistolu. Thank you for your time going through my pr and suggesting helpful changes.

Reply:
> Hi @Whoistolu. Thank you for your time going through my pr and suggesting helpful changes.

You are welcome @ezraermy 🥇